atuin/src/command/client/search/engines/skim.rs

```diff
@@ -54,6 +54,10 @@ async fn fuzzy_search(
             yield_now().await;
         }
         let context = &state.context;
+        let git_root = context.git_root.clone().map_or_else(
+            || context.cwd.clone(),
+            |git_root| git_root.to_str().unwrap_or("/").to_string(),
+        );
```

Comment:
I'd like to avoid allocations in the search loop.

I believe this should compile
```suggestion
        let git_root = context.git_root.as_ref().and_then(|git_root| git_root.to_str()).unwrap_or(&context.cwd);
```